Website/Packages/datax-pipeline/src/modules/flowDefinition/components/schedule/recurringScheduleSettings.jsx

```diff
@@ -0,0 +1,282 @@
+        return (
+            <div style={batchTypeSection}>
+                <TextField
+                    className="ms-font-m info-settings-textbox"
+                    spellCheck={false}
+                    label="Batch Type"
+                    disabled={true}
+                    value={this.props.batchTypeDisplayName}
+                />
+            </div>
+        );
+    }
+
+    renderTimeRange() {
+        return (
+            <div >
+                {this.renderStartTime()}
+                {this.renderEndTime()}
+            </div>
+        );
+    }
+    
+    renderStartTime() {
+        return (
+            <div style={sectionStyle}>
+                <Label className="ms-font-m info-settings-textbox">Start Time</Label>
+                <DateTimePicker
+                    className="ms-font-m info-settings-textbox"
+                    value={this.props.batch.properties.startTime}
+                    onChange={(value) => this.props.onUpdateBatchStartTime(value)}
+                />
+            </div>
+        );
+    }
+
+    renderEndTime() {
+        return (
+            <div style={sectionStyle}>
+                <Label className="ms-font-m info-settings-textbox">End Time</Label>
+                <DateTimePicker
+                    className="ms-font-m info-settings-textbox"
+                    value={this.props.batch.properties.endTime}
+                    onChange={(value) => this.props.onUpdateBatchEndTime(value)}
+                />
+            </div>
+        );
+    }
+
+    renderInterval() {
+        return (
+            <div style={sectionContainerStyle}>
+                {this.renderIntervalText()}
+                {this.renderIntervalTypeDropdown()}
+            </div>
+        );
+    }
+
+    renderIntervalText() {
+        return (
+            <div style={sectionValueStyle}>
+                <TextField
+                    className="ms-font-m"
+                    spellCheck={false}
+                    label="Recurrence"
+                    value={this.props.batch.properties.interval}
+                    onChange={(event, value) => this.props.onUpdateBatchIntervalValue(value)}
+                />
+            </div>
+        );
+    }
+
+    renderIntervalTypeDropdown() {
+        const options = Models.batchIntervalTypes.map(type => {
+            return {
+                key: type.key,
+                text: type.name,
+                disabled: type.disabled
+            };
+        });
+
+        return (
+            <div style={sectionDropdownStyle}>
+                <Label className="ms-font-m">Unit</Label>
+                <Dropdown
+                    className="ms-font-m"
+                    options={options}
+                    selectedKey={this.props.batch.properties.intervalType}
+                    onChange={(event, selection) => this.props.onUpdateBatchIntervalType(selection.key)}
+                />
+            </div>
+        );
+    }
+
+    renderDelay() {
```

Comment:
please see my remarks in one time job settings. We don't need 3 functions duplicated here

Reply:
Done. created/used the helper methods.